### PR TITLE
Add unit test for core/crypto and a little change in the empty  []byte return value.

### PR DESCRIPTION
**Pull request by yigithankarabulut**
# Added unit testing to core/crypto/decrypt_test.go

## Description

Test coverage increased from %11.8 to %85.5 in core/crypto.
Updated the DecryptBytesAge() function to return nil instead of []bytes{} if it returns an error.

## Changes

List the major changes you have made in bullet points:

Add Test for core/crypto/decrypt_test.go #891

## Test Policy Compliance

- [x] I have added or updated unit tests for my changes.
- [ ] I have included integration tests where applicable.
- [x] All new and existing tests pass successfully.

## Code Quality

- [x] I have followed the coding standards for this project.
- [x] I have performed a self-review of my code.
- [x] My code is well-commented, particularly in areas that may be difficult 
      to understand.

## Documentation

- [ ] I have made corresponding changes to the documentation (if applicable).
- [ ] I have updated any relevant READMEs or wiki pages.

## Additional Comments

Include any additional comments or context about the PR here.

## Checklist

Before you submit this PR, please make sure:
- [x] You have read [the contributing guidelines][contributing] and 
      *especially* the [test policy][test-policy].
- [x] You have thoroughly tested your changes.
- [x] You have followed all the contributing guidelines for this project.
- [x] You understand and agree that your contributions will be publicly available 
  under the project's license.

[contributing]: https://vsecm.com/docs/contributing/
[test-policy]: https://vsecm.com/docs/contributing/#add-tests-for-new-features

*By submitting this pull request, you confirm that my contribution is made under 
the terms of the project's license and that you have the authority to grant 
these rights.*

---

Thank you for your contribution to [**VMware Secrets Manager**](https://vsecm.com)
🐢⚡️!

### CODE REVIEW

File: core/crypto/decrypt.go [hunk 1]
```diff
@@ -78,14 +78,14 @@ func DecryptBytesAge(data []byte) ([]byte, error) {
 
 	identity, err := age.ParseX25519Identity(privateKey)
 	if err != nil {
-		return []byte{}, errors.Join(
+		return nil, errors.Join(
```
Comment (v0lkan) [inline core/crypto/decrypt.go:81]: +1
